src/v/rpc/transport.h

```diff
@@ -167,10 +167,13 @@ ss::future<result<rpc::client_context<T>>> parse_result(
     // success case
     if (st == status::success) {
         return parse_type<T>(in, sctx->get_header())
-          .then([sctx = std::move(sctx)](T data) {
+          .then_wrapped([sctx = std::move(sctx)](ss::future<T> data_fut) {
+              if (data_fut.failed()) {
+                  sctx->body_parse_exception(data_fut.get_exception());
+              }
               sctx->signal_body_parse();
-              return ret_t(
-                rpc::client_context<T>(sctx->get_header(), std::move(data)));
+              return ret_t(rpc::client_context<T>(
+                sctx->get_header(), std::move(data_fut.get())));
```

Comment:
do you need to `return` before this in the case that `data_fut` is failed?

Reply:
very good point

Reply:
did you intend to push something related to this?

Reply:
i wasn't sure if any change were needed, but it looked like `get` would throw rather than returning a `result<T>` with an error.

Reply:
I've added a comment explaining why we want to throw that exception.